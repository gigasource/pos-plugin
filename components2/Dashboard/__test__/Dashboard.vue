<script>
import ContentRender from "../../../components/common/ContentRender";
import {KeepAlive, onActivated, ref, h} from "vue";
import {contentView} from "./dashboard-shared";
import _ from 'lodash';


const TextField = {
  setup(props) {
    const count = ref(0);
    const init = _.debounce(() => {
      count.value ++;
    }, 10);
    init();
    onActivated(() => {
      init();
      console.log('onActivated');
    })

    return () => <p>{count.value}</p>
  }
}

const TextComp = function () {
  return <p>c</p>
}

const views = {
  TextComp,
  TextField
}

export default {
  name: "Dashboard",
  components: {
    ContentRender,
  },
  setup() {
    return () => <KeepAlive>
      {h(views[contentView.value.name])}
    </KeepAlive>
  }
}
</script>

<style scoped>

</style>
