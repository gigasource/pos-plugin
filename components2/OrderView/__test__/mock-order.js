const mockOrder = {
  table: "10",
  _id: "5fee6600612d45b5ffdcabb7",
  items:
    [{
      groupPrinter: "Bar",
      id: "48",
      name: "Whiskey",
      price: 40,
      product: "5e784d7786e3ec0997b5e39b",
      taxes: [19, 7],
      course: 1,
      quantity: 1,
      _id: "5fee6600612d45b5ffdcabb8",
      vSum: 40,
      vTakeAway: false,
      tax: 19,
      sent: true,
      printed: true,
      vTaxSum: {
        "19": {
          tax: 6.39,
          net: 33.61,
          gross: 40
        }
      }
    }],
  cancellationItems: [],
  takeAway: false,
  status: "inProgress",
  payment: [],
  user: [],
  vSum: 40,
  vDate: "2020-12-31T17:00:00.000Z",
  vTaxSum: {
    '19': {
      tax: 6.39,
      net: 33.61,
      gross: 40
    }
  }
}

module.exports = {
  mockOrder
}
