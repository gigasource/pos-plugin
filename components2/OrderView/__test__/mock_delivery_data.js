import { mockProducts } from './mock_products'
import _ from 'lodash'

export const mockFavorites = _.filter(mockProducts, p => p.option && p.option.favorite)
export const selectedCustomerMock = {
  'name': 'Tony',
  'phone': '0988.888.888',
  'addresses': [
    {
      address: 'address val 1',
      zipcode: 'zipcode val 1',
      house: 'hose val 1',
      street: 'street val 1',
      city: 'city val 1'
    },
    {
      address: 'address val 2',
      zipcode: 'zipcode val 2',
      house: 'hose val 2',
      street: 'street val 2',
      city: 'city val 2'
    }
  ],
}

export default {
  'favorites': mockFavorites, // favorite products
  //
  'selectedCustomer': selectedCustomerMock,

  // index of address will be used to ship product (in case the selected customer have more than 1 addrs)
  // TODO: [consider] rename to selectedAddressIndex
  'selectedAddress': 0,

  'address': '', // show in tablet mode for new customer
  'dialogMode': 'add', // edit
  'autocompleteAddresses': [], // only show in mobile mode (gather by query to cms backend)
  'placeId': '', // show in dialog input slot & deliveryOrderMode='mobile'
  'showKeyboard': false, // true
  'deliveryOrderMode': 'tablet', // mobile

  //
  'calls': [],
  'missedCalls': [],

  // doesn't show in snapshot (using in non-default slots)
  'name': 'Tommy',
  'phone': '084.999.8888',
  'zipcode': '',
  'city': '',
  'house': '',
  'street': '',
}
