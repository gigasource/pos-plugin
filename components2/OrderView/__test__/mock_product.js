export const mockProduct = {
  "_id": "5e784d7786e3ec0997b5e39b",
  "category": {
    "_id": "5df6f5dd7854273fa4fd989e",
    "name": "Wine",
    "position": 5
  },
  "groupPrinter": {
    "_id": "5e7864b186e3ec0997b5e48e",
    "name": "Bar",
    "printer": "HP M176n",
    "printers": [
      {
        "_id": "5e81774685fb2d03dda95cd0",
        "printerType": "ip",
        "ip": "192.168.10.6",
        "escPOS": false,
        "merge": "true"
      }
    ],
    "showDineInTax": false,
    "type": "kitchen"
  },
  "id": "48",
  "layouts": [
    {
      "_id": "5e784d7786e3ec0997b5e39c",
      "color": "#DCE775",
      "order": 2,
      "device": "pos"
    }
  ],
  "name": "Whiskey",
  "option": {},
  "price": 40,
  "tax": 19,
  "tax2": 7
}
