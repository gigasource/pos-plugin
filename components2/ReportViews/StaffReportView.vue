<script>
import dayjs from 'dayjs';
import {onActivated, ref} from 'vue';
import {$filters} from '../AppSharedStates';
import {useI18n} from 'vue-i18n';
import {GIcon, GTabItem, GTabs, GToolbar} from '../../../../backoffice/pos-vue-framework';
import {backFn, dateFormat, genScopeId, timeFormat} from "../utils";
import {
  getListUsers,
  getStaffReport,
  groupByPayment, printStaffReport,
  reportDate,
  selectedStaff,
  staffs,
  userSales
} from "./staf-report-logic";
import _ from 'lodash';

export default {
  components: [GTabs, GTabItem, GToolbar, GIcon],
  setup() {
    const {t} = useI18n()

    //todo: printStaffReport
    async function print() {
      return await printStaffReport()
    }

    function getFormattedTime(val) {
      return val ? dayjs(val).format(`${dateFormat.value} ${timeFormat.value}`) : ''
    }

    onActivated(async () => {
      await getListUsers();
      await getStaffReport();
    })
    getListUsers()
    getStaffReport();

    const genSalesDetail = (user) => {
      return <>
        {
          userSales.value && userSales.value[user] &&
          <div>
            <div class="detail-header">
              {t('report.reportDate')}: {reportDate.value} </div>
            <div class="detail-time">
              {t('report.firstOrder')}:
              {getFormattedTime(userSales.value[user].from)}
            </div>
            <div class="detail-time">
              {t('report.lastOrder')}:
              {getFormattedTime(userSales.value[user].to)}
            </div>
          </div>
        }
        <div class="sales-details-header">
          {t('common.sales')}
        </div>
        {
          userSales.value && userSales.value[user] &&
          <div>
            <p>
              <span class="sales-entry sales-type">
                {t('common.total')}
              </span>
              <span class="sales-entry sales-amount">
                {$filters.formatCurrency(userSales.value[user].vTaxSum.gross)}
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                {t('common.subtotal')}
              </span>
              <span class="sales-entry sales-amount">
                {$filters.formatCurrency(userSales.value[user].vTaxSum.net)}
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                {t('common.tax')} </span>
              <span class="sales-entry sales-amount">
                {$filters.formatCurrency(userSales.value[user].vTaxSum.tax)}
              </span>
            </p>
          </div>
        }
      </>
    }

    const genTaxesDetail = (user) => {
      return <>
        <div className="tax-detail">
          {
            userSales.value && userSales.value[user] && userSales.value[user].vTaxSum &&
            <div>
              {_.map(userSales.value[user].vTaxSum.vTaxSum, (taxObject, taxPercent, index) =>
                  <div key={index}>
                    <p className="sales-entry sales-type">
                      {t('common.tax')} {taxPercent}%:
                    </p>
                    <p>
                      <span className="sales-entry sales-type">
                        {t('common.total')}
                      </span>
                      <span className="sales-entry sales-amount">
                        {$filters.formatCurrency(taxObject.gross)}
                      </span>
                    </p>
                    <p>
                      <span className="sales-entry sales-type">
                        {t('common.subtotal')}
                      </span>
                      <span className="sales-entry sales-amount">
                        {$filters.formatCurrency(taxObject.net)}
                      </span>
                    </p>
                    <p>
                      <span className="sales-entry sales-type">
                        {t('common.tax')}
                      </span>
                      <span className="sales-entry sales-amount">
                        {$filters.formatCurrency(taxObject.tax)}
                      </span>
                    </p>
                    <br> </br>
                  </div>
              )}
            </div>
          }
          {
            userSales.value && userSales.value[user] &&
            <div>
              <p>
                <span className="sales-entry sales-type">
                  {t('report.vouchersSold')}
                </span>
                <span className="sales-entry sales-amount">
                  {$filters.formatCurrency(0)}
                </span>
              </p>
              <p>
                <span className="sales-entry sales-type">
                  {t('report.vouchersUsed')}
                </span>
                <span className="sales-entry sales-amount">
                  {$filters.formatCurrency(0)}
                </span>
              </p>
              <p>
                <span className="sales-entry sales-type">
                  {t('common.discount')}
                </span>
                <span className="sales-entry sales-amount">
                  {
                    $filters.formatCurrency(userSales.value[user].vDiscount)
                  }
                </span>
              </p>
            </div>
          }
        </div>
      </>
    }

    //todo: voucher
    //todo: cancellation
    const genPayment = (user) => {
      return groupByPayment.value && groupByPayment.value[user] &&
          <div class="sales-details">
            {_.map(groupByPayment.value[user], (sale, paymentType, index) =>
                <p key={index}>
                  <span class="sales-entry sales-type">
                    {paymentType} {t('common.sales')}:
                  </span>
                  <span class="sales-entry">
                    {$filters.formatCurrency(sale)}
                  </span>
                </p>
            )}<p>
            <span class="sales-entry sales-type">
              {t('report.returnedTotal')}:
            </span>
            <span class="sales-entry">
              {$filters.formatCurrency(0)}
            </span>
          </p>
          </div>
    }

    return genScopeId(() => (
        <div class="staff-report-view">
          <div class="staff-report-content">
            {staffs.value.length &&
            <g-tabs items={staffs.value} showArrows={false}
                    vertical={true} color="#F2F2F2"
                    slider-color="#2979FF" slider-size="4px"
                    text-color="#000000" v-model={selectedStaff.value}
                    v-slots={{
                      default: () => staffs.value.map((item, i) =>
                          <g-tab-item item={item} key={i}>
                            <div class="detail-header">
                              {t('report.staffName')}: {item.name}
                            </div>
                            {genSalesDetail(item.name)}
                            {genTaxesDetail(item.name)}
                            {genPayment(item.name)}
                          </g-tab-item>
                      ),
                      tab: ({item, index}) =>
                          <g-tab item={item} key={index} active-text-color="#000000">
                            <p class="tab-username">
                              {item.name}
                            </p>
                          </g-tab>
                    }}/>
            }
          </div>
          <g-toolbar bottom color="#EEEEEE">
            <g-btn uppercase={false} onClick={backFn.value} width="94px">
              <g-icon class="mr-2" svg>icon-back</g-icon>
              {t('ui.back')}
            </g-btn>
            <g-spacer></g-spacer>
            <g-btn uppercase={false} onClick={print} background-color="#2979FF" text-color="#FFFFFF">
              <g-icon class="mr-2" svg>icon-print2</g-icon>
              {t('report.printReport')}
            </g-btn>
          </g-toolbar>
        </div>
    ))
  }
}
</script>

<style lang="scss" scoped>
.staff-report-view {
  width: 100%;
  background-color: #fff;
  position: relative;
  height: 100%;

  .staff-report-content {
    height: 100%;

    .detail-header {
      font-weight: bold;
      font-size: 16px;
      line-height: 27px;
    }

    .detail-time {
      font-size: 14px;
      line-height: 25px;
      font-family: "Muli", sans-serif;
    }

    .sales-details {
      border-top: 1px dashed #000000;
    }

    .sales-details-header {
      font-weight: bold;
      font-size: 16px;
      line-height: 33px;
      border-top: 1px dashed #000000;
    }

    .tax-detail {
      border-top: 1px dashed #000000;
    }

    .sales-entry {
      font-size: 14px;
      line-height: 25px;
    }

    .sales-type {
      text-transform: capitalize;
    }

    .sales-amount {
      float: right;
    }

    .g-tab {
      height: 99px;
      width: 190px;
      background-color: #ffffff;
    }

    .g-tab:hover:before {
      opacity: 0;
    }

    .g-tab:not(.g-tab__active) {
      height: 99px;
      width: 190px;
      background-color: #F9F9F9;
    }

    .tab-username {
      text-transform: uppercase;
      white-space: normal;
    }

    ::v-deep .g-tabs-wrapper {
      height: calc(100% - 64px);

      .g-tabs__vertical {
        .g-tabs-bar {
          overflow-y: auto;
          -ms-overflow-style: none;
        }

        /* Hide scrollbar for Chrome, Safari and Opera */
        .g-tabs-bar::-webkit-scrollbar {
          display: none;
        }
      }

      .g-tab-items {
        height: 100%;
        overflow-y: auto;

        .g-tab-item {
          padding: 40px 50px;
          background-color: #ffffff;
        }

      }

      .g-tabs-slider {
        z-index: 2;
      }
    }

  }
}
</style>
