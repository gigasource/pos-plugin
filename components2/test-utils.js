import "../jest.setup";
import _ from 'lodash';
import {config, mount} from "@vue/test-utils";
import {hooks} from "./OrderView/pos-logic";
import {disablePay, payBtnClickable, payPrintMode, showIcon} from "./OrderView/pos-logic-be";
import StubFactory from './StubFactory';

jest.mock("vue-i18n", () => {
  return {
    useI18n() {
      return {
        t(t) {
          return t; // [Consider] return `t('${t}')` to indicate that i18n has been used
        },
        locale: "en"
      };
    }
  };
});
jest.mock("vue-router", () => {
  return {
    useRoute() {
      return {
        query: {
          type: 'default'
        }
      };
    },
    useRouter() {
      return {
        push() {
        }
      };
    }
  };
});

const dayjs = require("dayjs");
export const orm = require("schemahandler");
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const {stringify} = require("schemahandler/utils");
//jest.useFakeTimers("modern").setSystemTime(new Date("2021-01-01").getTime());
require("mockdate").set(new Date("2021-01-01").getTime());

const delay = require("delay");

export const foodTax = {taxes: [5, 10]};
export const drinkTax = {taxes: [16, 32]};

export const cola = {name: "Cola", price: 1.3, quantity: 1, ...drinkTax};
export const fanta = {name: "Fanta", price: 2, quantity: 1, ...drinkTax};
export const rice = {name: "Rice", price: 10, quantity: 1, ...foodTax};
export const ketchup = {name: "Add Ketchup", price: 3, quantity: 1};

beforeAll(async () => {
  orm.connect({uri: "mongodb://localhost:27017"}, "myproject");
  orm.registerSchema("Order", {
    inProgress: Boolean,
    items: [{}],
    table: Number
  });
  await orm("Order").deleteMany();
  await orm("Commit").deleteMany();

  orm.plugin(syncPlugin);
  orm.plugin(syncFlow);
  orm.registerCommitBaseCollection("Order");
  orm.plugin(require("../backend/commit/orderCommit"));
  orm.registerCollectionOptions("Order");
  orm.emit("commit:flow:setMaster", true);
});

afterEach(async () => {
  await orm("Order")
    .deleteMany()
    .direct();
  await orm("Commit")
    .deleteMany()
    .direct();
});

export let actions = [];
["showOrderReceipt", "pay", "printOrder"].forEach(e =>
  hooks.on(e, () => actions.push(e))
);
export let expectArray = () => [
  "payBtnClickable",
  payBtnClickable.value,
  "payPrintMode",
  payPrintMode.value,
  "showIcon",
  showIcon.value,
  "disablePay",
  disablePay.value,
  actions
];
//</editor-fold>

//<editor-fold desc="component">
export let component, wrapper;

export function setComponent(_component) {
  component = _component;
}

/**
 *
 * @param _component
 * @param options
 * @param useDefaults indicate whether _.defaults will be use for options
 * @returns {*}
 * example: makeWrapper(Order2, {shallow: false});
 */
export const makeWrapper = (_component, options, useDefaults = false) => {
  const defaultOption = {
    props: {},
    shallow: true,
    global: {
      directives: {
        touch: true
      },
      stubs: {
        'choose-table-dialog': true,
        'dialog-change-value': true,
        'dialog-choose-popup-modifier': true,
        "dialog-config-order-item": true,
        'dialog-form-input': true,
        'dialog-multi-payment': true,
        'dialog-multi-payment2': true,
        'dialog-product-search-result': true,
        'dialog-text-filter': true,
        'g-avatar': true,
        'g-autocomplete': true,
        'g-badge': true,
        'g-btn': true,
        "g-btn-bs": true,
        'g-card': true,
        'g-card-title': true,
        'g-combobox': true,
        'g-dialog': true,
        'g-divider': true,
        "g-icon": true,
        'g-item-group': true,
        'g-menu': true,
        "g-overlay": true,
        'g-radio': true,
        'g-simple-table': true,
        'g-spacer': true,
        'g-switch': true,
        'g-table': true,
        'g-tabs': true,
        'g-tab-item': true,
        'g-text-field': true,
        'g-text-field-bs': true,
        'g-toolbar': true,
        'g-snackbar': true,
        'pos-order': true,
        'pos-quick-order-toolbar': true,
        //'pos-order-split-order': true,
        'pos-keyboard-full': true,
        'pos-order-keyboard': true,
        'pos-order-move-items': true,
        'pos-order-receipt': true,
        'pos-order-voucher-dialog': true,
        'pos-textfield-new': true,
        'g-grid-select': StubFactory('g-grid-select'),
        'pos-payment-screen-payment-methods2': true,
        'pos-payment-screen-keyboard2': true,
        'pos-restaurant-payment-order-detail2': true,
        'pos-restaurant-payment-toolbar2': true,
      },
      mocks: {
        t: a => a,
        $t: a => a,
        $filters: {
          formatCurrency(val, decimals = 2) {
            if (!val || isNaN(val) || Math.floor(val) === val) return val;
            return val.toFixed(decimals);
          }
        }
      }
    }
  }
  const optionResult = useDefaults ? _.defaults(options, defaultOption) : _.merge(defaultOption, options)
  wrapper = mount(component || _component, optionResult);
};

beforeAll(() => (config.renderStubDefaultSlot = true));
