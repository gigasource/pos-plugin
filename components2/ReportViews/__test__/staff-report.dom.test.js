//<editor-fold desc="declare">
import {makeWrapper, orm, wrapper} from "../../test-utils";

import {nextTick} from "vue";
import {demoData} from "../../OrderView/__test__/demoData";
import dayjs from "dayjs";
import customParseFormat from 'dayjs/plugin/customParseFormat';
import StaffReportView from "../StaffReportView";
import {currentDate, getListUsers, getStaffReport, printStaffReport, selectedStaff, staffs} from "../staf-report-logic";
import _ from 'lodash';

dayjs.extend(customParseFormat)

const {prepareDb, prepareReportTest} = require('../../../backend/order-logic/report.prepare.test')

const {stringify} = require("schemahandler/utils");

const delay = require("delay")

jest.mock('cms', () => {
  process.env.USE_GLOBAL_ORM = true
  const {cmsFactory} = require('../../../test-utils')
  const _cms = cmsFactory('eodDom')
  global.cms = _cms
  return {
    socket: _cms.feSocket,
    getModel: function (modelName) {
      return _cms.orm(modelName)
    }
  }
})

let cms = global.cms

//require('../../../backend/print/print-report/report-index')(cms);

//</editor-fold>

describe("eod test", function () {
  beforeAll(async () => {
    await cms.init()
    await prepareReportTest(cms)
    await require('../../../backend/print/print-report/report-index')(cms);

    await prepareDb(/*cms.*/orm)
    cms.triggerFeConnect()
  });

  it("case 0 getListUsers", async function () {
    await getListUsers();
    currentDate.value = dayjs('05.01.2021', 'DD.MM.YYYY').toDate();
    await getStaffReport();
  }, 80000);

  it("case 1 StaffReportView", async function () {
    currentDate.value = dayjs('05.01.2021', 'DD.MM.YYYY').toDate();

    makeWrapper(StaffReportView, {
      //shallow: true,
      props: {}
    });

    //await getMonthReport();
    await delay(200);
    await nextTick();

    selectedStaff.value = _.find(staffs.value, {name: 'Waiter 2'});
    await delay(200);
    await nextTick();
    printStaffReport();
    await delay(200);
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);

});
