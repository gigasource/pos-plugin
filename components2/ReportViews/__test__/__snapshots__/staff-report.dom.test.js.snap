// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eod test case 1 StaffReportView 1`] = `
<div class="staff-report-view">
  <div class="staff-report-content">
    <g-tabs-stub items="[object Object],[object Object],[object Object],[object Object]"
                 showarrows="false"
                 vertical="true"
                 color="#F2F2F2"
                 slider-color="#2979FF"
                 slider-size="4px"
                 text-color="#000000"
                 modelvalue="[object Object]"
    >
      <g-tab-item-stub item="[object Object]">
        <div class="detail-header">
          report.staffName: admin
        </div>
        <div class="sales-details-header">
          common.sales
        </div>
        <div class="tax-detail">
        </div>
      </g-tab-item-stub>
      <g-tab-item-stub item="[object Object]">
        <div class="detail-header">
          report.staffName: Manager
        </div>
        <div class="sales-details-header">
          common.sales
        </div>
        <div class="tax-detail">
        </div>
      </g-tab-item-stub>
      <g-tab-item-stub item="[object Object]">
        <div class="detail-header">
          report.staffName: Waiter 1
        </div>
        <div>
          <div class="detail-header">
            report.reportDate: 05.01.2021
          </div>
          <div class="detail-time">
            report.firstOrder:05.01.2021 00:00
          </div>
          <div class="detail-time">
            report.lastOrder:05.01.2021 00:00
          </div>
        </div>
        <div class="sales-details-header">
          common.sales
        </div>
        <div>
          <p>
            <span class="sales-entry sales-type">
              common.total
            </span>
            <span class="sales-entry sales-amount">
              6.60
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              common.subtotal
            </span>
            <span class="sales-entry sales-amount">
              5.55
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              common.tax
            </span>
            <span class="sales-entry sales-amount">
              1.05
            </span>
          </p>
        </div>
        <div class="tax-detail">
          <div>
            <div>
              <p class="sales-entry sales-type">
                common.tax 16%:
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.total
                </span>
                <span class="sales-entry sales-amount">
                  5.30
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.subtotal
                </span>
                <span class="sales-entry sales-amount">
                  4.57
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.tax
                </span>
                <span class="sales-entry sales-amount">
                  0.73
                </span>
              </p>
              <br>
            </div>
            <div>
              <p class="sales-entry sales-type">
                common.tax 32%:
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.total
                </span>
                <span class="sales-entry sales-amount">
                  1.30
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.subtotal
                </span>
                <span class="sales-entry sales-amount">
                  0.98
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.tax
                </span>
                <span class="sales-entry sales-amount">
                  0.32
                </span>
              </p>
              <br>
            </div>
          </div>
          <div>
            <p>
              <span class="sales-entry sales-type">
                report.vouchersSold
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                report.vouchersUsed
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                common.discount
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
          </div>
        </div>
        <div class="sales-details">
          <p>
            <span class="sales-entry sales-type">
              cash common.sales:
            </span>
            <span class="sales-entry">
              6.60
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              report.returnedTotal:
            </span>
            <span class="sales-entry">
              0
            </span>
          </p>
        </div>
      </g-tab-item-stub>
      <g-tab-item-stub item="[object Object]">
        <div class="detail-header">
          report.staffName: Waiter 2
        </div>
        <div>
          <div class="detail-header">
            report.reportDate: 05.01.2021
          </div>
          <div class="detail-time">
            report.firstOrder:05.01.2021 00:00
          </div>
          <div class="detail-time">
            report.lastOrder:05.01.2021 00:00
          </div>
        </div>
        <div class="sales-details-header">
          common.sales
        </div>
        <div>
          <p>
            <span class="sales-entry sales-type">
              common.total
            </span>
            <span class="sales-entry sales-amount">
              14.30
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              common.subtotal
            </span>
            <span class="sales-entry sales-amount">
              13.50
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              common.tax
            </span>
            <span class="sales-entry sales-amount">
              0.80
            </span>
          </p>
        </div>
        <div class="tax-detail">
          <div>
            <div>
              <p class="sales-entry sales-type">
                common.tax 5%:
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.total
                </span>
                <span class="sales-entry sales-amount">
                  13
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.subtotal
                </span>
                <span class="sales-entry sales-amount">
                  12.38
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.tax
                </span>
                <span class="sales-entry sales-amount">
                  0.62
                </span>
              </p>
              <br>
            </div>
            <div>
              <p class="sales-entry sales-type">
                common.tax 16%:
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.total
                </span>
                <span class="sales-entry sales-amount">
                  1.30
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.subtotal
                </span>
                <span class="sales-entry sales-amount">
                  1.12
                </span>
              </p>
              <p>
                <span class="sales-entry sales-type">
                  common.tax
                </span>
                <span class="sales-entry sales-amount">
                  0.18
                </span>
              </p>
              <br>
            </div>
          </div>
          <div>
            <p>
              <span class="sales-entry sales-type">
                report.vouchersSold
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                report.vouchersUsed
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
            <p>
              <span class="sales-entry sales-type">
                common.discount
              </span>
              <span class="sales-entry sales-amount">
                0
              </span>
            </p>
          </div>
        </div>
        <div class="sales-details">
          <p>
            <span class="sales-entry sales-type">
              card common.sales:
            </span>
            <span class="sales-entry">
              2
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              cash common.sales:
            </span>
            <span class="sales-entry">
              12.30
            </span>
          </p>
          <p>
            <span class="sales-entry sales-type">
              report.returnedTotal:
            </span>
            <span class="sales-entry">
              0
            </span>
          </p>
        </div>
      </g-tab-item-stub>
    </g-tabs-stub>
  </div>
  <g-toolbar-stub bottom="true"
                  color="#EEEEEE"
  >
    <g-btn-stub uppercase="false"
                width="94px"
    >
      <g-icon-stub class="mr-2"
                   svg="true"
      >
        icon-back
      </g-icon-stub>
      ui.back
    </g-btn-stub>
    <g-spacer-stub>
    </g-spacer-stub>
    <g-btn-stub uppercase="false"
                background-color="#2979FF"
                text-color="#FFFFFF"
    >
      <g-icon-stub class="mr-2"
                   svg="true"
      >
        icon-print2
      </g-icon-stub>
      report.printReport
    </g-btn-stub>
  </g-toolbar-stub>
</div>
`;
